pkg/v1/remote/write.go

```diff
@@ -58,16 +57,16 @@ func Write(ref name.Reference, img v1.Image, auth authn.Authenticator, t http.Ro
 
 	// Upload individual layers in goroutines and collect any errors.
 	// If we can dedupe by the layer digest, try to do so. If the layer is
-	// a stream.Layer, we can't dedupe and might re-upload.
+	// a streaming layer, we can't dedupe and might re-upload.
```

Comment:
This comment could be even more general. "If we can't determine the digest to dedupe for whatever reason..."